test/unit_test/app/data_client_test.dart

```diff
@@ -8,12 +8,78 @@ import 'package:mockito/mockito.dart';
 import 'package:viam_sdk/protos/app/data.dart';
 import 'package:viam_sdk/protos/app/data_sync.dart' hide CaptureInterval;
 import 'package:viam_sdk/src/app/data.dart';
+import 'package:viam_sdk/src/gen/app/data/v1/data.pb.dart';
+import 'package:viam_sdk/src/gen/app/data/v1/data.pbgrpc.dart';
+import 'package:viam_sdk/src/gen/app/data/v1/data.pbjson.dart';
 import 'package:viam_sdk/src/media/image.dart';
 import 'package:viam_sdk/src/utils.dart';
 
 import '../mocks/mock_response_future.dart';
 import '../mocks/service_clients_mocks.mocks.dart';
 
+class FakeDataServiceClient extends MockDataServiceClient {
+  bool tagsAddedByIds = false;
+  bool tagsAddedByFilter = false;
+  bool removeBboxIdCalled = false;
+  bool configureDatabaseUserCalled = false;
+  bool addBinaryDataToDatasetByIdsCalled = false;
+  bool removeBinaryDataFromDatasetByIdsCalled = false;
```

Comment:
You don't need these, you should be able to use `mockito` to confirm that these functions were called: https://pub.dev/packages/mockito#verifying-exact-number-of-invocations--at-least-x--never

Reply:
Which i think means that this entire fake client can be removed

Reply:
Oh fantastic, thanks!